addons/html_editor/static/src/main/list/list_plugin.js

```diff
@@ -70,6 +70,7 @@ export class ListPlugin extends Plugin {
                 title: _t("Checklist"),
                 description: _t("Track tasks with a checklist"),
                 icon: "fa-check-square-o",
+                isDisabled: this.config.disableCheckbox,
```

Comment:
Can't we just use isAvailable and disable the button creation when isAvailable returns false?